diff --git a/src/process/exec.mach b/src/process/exec.mach
index d763841..22dcf35 100644
--- a/src/process/exec.mach
+++ b/src/process/exec.mach
@@ -829,10 +829,16 @@ $or {
         ret buf;
     }
 
-    fun prog_ok() str { ret "/bin/true"; }
-    fun argv_ok() [4]usize { ret make_argv1("/bin/true"); }
-    fun prog_fail() str { ret "/bin/false"; }
-    fun argv_fail() [4]usize { ret make_argv1("/bin/false"); }
+    fun prog_ok() str {
+        $if ($mach.build.os == $mach.os.darwin) { ret "/usr/bin/true"; }
+        $or { ret "/bin/true"; }
+    }
+    fun argv_ok() [4]usize { ret make_argv1(prog_ok()); }
+    fun prog_fail() str {
+        $if ($mach.build.os == $mach.os.darwin) { ret "/usr/bin/false"; }
+        $or { ret "/bin/false"; }
+    }
+    fun argv_fail() [4]usize { ret make_argv1(prog_fail()); }
     fun prog_echo() str { ret "/bin/echo"; }
     fun argv_echo() [4]usize { ret make_argv2("/bin/echo", "hello"); }
 
